# How do I change display options in pandas?

🐼 Tuto on pandas by Data School - Exercice performed by Dorian.H Mekni 🥇 | Thu 17 Dec 2020

In [1]:
import pandas as pd 

In [3]:
drinks = pd.read_csv('http://bit.ly/drinksbycountry')

In [4]:
drinks

country  beer_servings  spirit_servings  wine_servings  \
0    Afghanistan              0                0              0   
1        Albania             89              132             54   
2        Algeria             25                0             14   
3        Andorra            245              138            312   
4         Angola            217               57             45   
..           ...            ...              ...            ...   
188    Venezuela            333              100              3   
189      Vietnam            111                2              1   
190        Yemen              6                0              0   
191       Zambia             32               19              4   
192     Zimbabwe             64               18              4   

     total_litres_of_pure_alcohol      continent  
0                             0.0           Asia  
1                             4.9         Europe  
2                             0.7         Africa  
3                            12.4         Europe  
4                             5.9         Africa  
..                            ...            ...  
188                           7.7  South America  
189                           2.0           Asia  
190                           0.1           Asia  
191                           2.5         Africa  
192                           4.7         Africa  

[193 rows x 6 columns]


☝🏻 From our first glance, we notice that there are 193 rows. We are being shown the first 5 rows and the last 5 ones. 



⭐️  Now, what if I want to read through all the rows : 
    

In [5]:
# Top level function use : 
pd.get_option('display.max_rows')

60

In [ ]:
☝🏻 Usually what you could be shown as a default value meaning 30 rows at the top and 30 rows at the bottom. 

In [8]:
# Let's change this : 
pd.set_option('display.max_rows', None)

In [9]:
# Let's check if it worked : 
drinks 

country  beer_servings  spirit_servings  \
0                     Afghanistan              0                0   
1                         Albania             89              132   
2                         Algeria             25                0   
3                         Andorra            245              138   
4                          Angola            217               57   
5               Antigua & Barbuda            102              128   
6                       Argentina            193               25   
7                         Armenia             21              179   
8                       Australia            261               72   
9                         Austria            279               75   
10                     Azerbaijan             21               46   
11                        Bahamas            122              176   
12                        Bahrain             42               63   
13                     Bangladesh              0                0   
14                       Barbados            143              173   
15                        Belarus            142              373   
16                        Belgium            295               84   
17                         Belize            263              114   
18                          Benin             34                4   
19                         Bhutan             23                0   
20                        Bolivia            167               41   
21             Bosnia-Herzegovina             76              173   
22                       Botswana            173               35   
23                         Brazil            245              145   
24                         Brunei             31                2   
25                       Bulgaria            231              252   
26                   Burkina Faso             25                7   
27                        Burundi             88                0   
28                  Cote d'Ivoire             37                1   
29                     Cabo Verde            144               56   
30                       Cambodia             57               65   
31                       Cameroon            147                1   
32                         Canada            240              122   
33       Central African Republic             17                2   
34                           Chad             15                1   
35                          Chile            130              124   
36                          China             79              192   
37                       Colombia            159               76   
38                        Comoros              1                3   
39                          Congo             76                1   
40                   Cook Islands              0              254   
41                     Costa Rica            149               87   
42                        Croatia            230               87   
43                           Cuba             93              137   
44                         Cyprus            192              154   
45                 Czech Republic            361              170   
46                    North Korea              0                0   
47                       DR Congo             32                3   
48                        Denmark            224               81   
49                       Djibouti             15               44   
50                       Dominica             52              286   
51             Dominican Republic            193              147   
52                        Ecuador            162               74   
53                          Egypt              6                4   
54                    El Salvador             52               69   
55              Equatorial Guinea             92                0   
56                        Eritrea             18                0   
57                      


✅ By changing the default value, we get to read the entire 193 columns.


In [10]:
# Let's reset to what it was previosuly : 
pd.reset_option('display.max_rows')

In [11]:
# Let's check it worked : 
drinks

country  beer_servings  spirit_servings  wine_servings  \
0    Afghanistan              0                0              0   
1        Albania             89              132             54   
2        Algeria             25                0             14   
3        Andorra            245              138            312   
4         Angola            217               57             45   
..           ...            ...              ...            ...   
188    Venezuela            333              100              3   
189      Vietnam            111                2              1   
190        Yemen              6                0              0   
191       Zambia             32               19              4   
192     Zimbabwe             64               18              4   

     total_litres_of_pure_alcohol      continent  
0                             0.0           Asia  
1                             4.9         Europe  
2                             0.7         Africa  
3                            12.4         Europe  
4                             5.9         Africa  
..                            ...            ...  
188                           7.7  South America  
189                           2.0           Asia  
190                           0.1           Asia  
191                           2.5         Africa  
192                           4.7         Africa  

[193 rows x 6 columns]


✅ Job done ! 


In [12]:
# Let's see now with another option : 
pd.get_option('display.max_columns')

20

☝🏻 Same scenario here, especially if you had more columns and that you wanted to read it all or less, you would set the parameter accordingly. 


In [17]:
# Let's use a different dataframe : 
train = pd.read_csv('http://bit.ly/kaggletrain')
train.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S


☝🏻 You would notice that on the name column, the title of the the passenger on row 1 has his name cut and then not fully readable. 

🧐 It is because pandas has a max limit of character a string can display within a dataframe. 



⭐️ Let's change that as we wish to read the name of this passenger in full : 


In [18]:
# Let's read how many characters a string can display : 
pd.get_option('display.max_colwidth')

50

In [19]:
# Let's change that to a specific number of characters although we could use None again : 
pd.set_option('display.max_colwidth', 1000)

In [20]:
# Reading through :  
train.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Thayer)  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S


✅ We can now read through the entire name ! 



⭐️ Let's now change the number of decimal points read within the Fare column, modifying the float values : 


In [23]:
# Reading what is the maximum decimal value a fload can be read through within a dataframe : 
pd.get_option('display.precision')

6

In [24]:
# Let's change that to 2 : 
pd.set_option('display.precision', 2)

In [26]:
# Checking time :
train.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Thayer)  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket   Fare Cabin Embarked  
0      0         A/5 21171   7.25   NaN        S  
1      0          PC 17599  71.28   C85        C  
2      0  STON/O2. 3101282   7.92   NaN        S  
3      0            113803  53.10  C123        S  
4      0            373450   8.05   NaN        S


✅ Job done ! 



🧐 To be clear, it did not affect the underlying data ! It only affected what was displayed. 



⭐️ How do we format large numbers so that they have commas when we're printing or graphing : 


In [ ]:
# Using the drinks dataframe to answer that question : 
drinks.head()

In [29]:
# No large numbers here so we'll add some columns that contain those : 
drinks['x'] = drinks.wine_servings * 1000 
drinks['y'] = drinks.total_litres_of_pure_alcohol * 1000

In [30]:
# Reading : 
drinks.head()

country  beer_servings  spirit_servings  wine_servings  \
0  Afghanistan              0                0              0   
1      Albania             89              132             54   
2      Algeria             25                0             14   
3      Andorra            245              138            312   
4       Angola            217               57             45   

   total_litres_of_pure_alcohol continent       x        y  
0                           0.0      Asia       0      0.0  
1                           4.9    Europe   54000   4900.0  
2                           0.7    Africa   14000    700.0  
3                          12.4    Europe  312000  12400.0  
4                           5.9    Africa   45000   5900.0

In [31]:
# Let's use format string -- something we use in python and that pandas did not invent : 
pd.set_option('display.float_format', '{:,}'.format)

In [32]:
# Reading through : 
drinks.head()

country  beer_servings  spirit_servings  wine_servings  \
0  Afghanistan              0                0              0   
1      Albania             89              132             54   
2      Algeria             25                0             14   
3      Andorra            245              138            312   
4       Angola            217               57             45   

   total_litres_of_pure_alcohol continent       x        y  
0                           0.0      Asia       0      0.0  
1                           4.9    Europe   54000  4,900.0  
2                           0.7    Africa   14000    700.0  
3                          12.4    Europe  312000 12,400.0  
4                           5.9    Africa   45000  5,900.0


☝🏻 Why did it affect y and not x ❓


In [33]:
# Reading through the dtypes : 
drinks.dtypes

country                          object
beer_servings                     int64
spirit_servings                   int64
wine_servings                     int64
total_litres_of_pure_alcohol    float64
continent                        object
x                                 int64
y                               float64
dtype: object


🧐 The reason why it happened is because y is a float and x is an int. There is indeed no int format but only a float format that affect numeric values of dtype float. 


😊 In the event, we would want to operate this change to x, we would convert the data type from int to float and then apply the same method to insert commas as done on the y column it weren't already done.  


# 🎩 Bonus tips : iPython | Jupyter Notebook ONLY

🤠 Let's say you're not connected to the internet, or you don't want to perform another google search to the internet to read the pandas documentation. You want to read up on the pandas options. How can you do that ? 

In [34]:
# Here comes a cool function that does just that : 
pd.describe_option()

compute.use_bottleneck : bool
    Use the bottleneck library to accelerate if it is installed,
    the default is True
    Valid values: False,True
    [default: True] [currently: True]compute.use_numexpr : bool
    Use the numexpr library to accelerate computation if it is installed,
    the default is True
    Valid values: False,True
    [default: True] [currently: True]display.chop_threshold : float or None
    if set to a float value, all float values smaller then the given threshold
    will be displayed as exactly 0 by repr and friends.
    [default: None] [currently: None]display.colheader_justify : 'left'/'right'
    Controls the justification of column headers. used by DataFrameFormatter.
    [default: right] [currently: right]display.column_space No description available.
    [default: 12] [currently: 12]display.date_dayfirst : boolean
    When True, prints and parses dates with the day first, eg 20/01/2005
    [default: False] [currently: False]display.date_yearfirst : bool

In [35]:
# A more refined way to target a specific option : rows : 
pd.describe_option('rows')

display.max_info_rows : int or None
    df.info() will usually show null-counts for each column.
    For large frames this can be quite slow. max_info_rows and max_info_cols
    limit this null check only to frames with smaller dimensions than
    specified.
    [default: 1690785] [currently: 1690785]display.max_rows : int
    If max_rows is exceeded, switch to truncate view. Depending on
    `large_repr`, objects are either centrally truncated or printed as
    a summary view. 'None' value means unlimited.

    In case python/IPython is running in a terminal and `large_repr`
    equals 'truncate' this can be set to 0 and pandas will auto-detect
    the height of the terminal and print a truncated object which fits
    the screen height. The IPython notebook, IPython qtconsole, or
    IDLE do not run in a terminal and hence it is not possible to do
    correct auto-detection.
    [default: 60] [currently: 60]display.min_rows : int
    The numbers of rows to show in a truncated view (wh

In [36]:
# A quick way to reset all your option : 
pd.reset_option('all')


: boolean
    use_inf_as_null had been deprecated and will be removed in a future
    version. Use `use_inf_as_na` instead.



/Users/mydesk/opt/anaconda3/lib/python3.7/site-packages/pandas/_config/config.py:622: FutureWarning: 
: boolean
    use_inf_as_null had been deprecated and will be removed in a future
    version. Use `use_inf_as_na` instead.

  warnings.warn(d.msg, FutureWarning)



☝🏻 You get a warning because some of the options has already been deprecated and you can safely ignore any warning in this case. 


🙏🏻 Thank you !

👋🏻 See you in the next one !